tests/plugins/test_resolve_composes.py

```diff
@@ -289,24 +289,45 @@ def test_request_compose_for_modules(self, workflow, reactor_config_map):  # noq
 
     @pytest.mark.parametrize(('pulp_arches', 'arches', 'signing_intent', 'expected_intent'), (  # noqa:F811
         (None, None, 'unsigned', 'unsigned'),
-        (['x86_64'], None, 'release', 'beta'),
+        # For the next test, since arches is none, no compose is performed even though pulp_arches
+        # has a value. Expected intent doesn't change when nothing is composed.
+        (['x86_64'], None, 'release', 'release'),
+        # pulp composes have the beta signing intent and downgrade the release intent to beta.
         (['x86_64'], ['x86_64'], 'release', 'beta'),
         (['x86_64', 'ppce64le'], ['x86_64', 'ppce64le'], 'release', 'beta'),
+        (['x86_64', 'ppce64le'], ['ppce64le'], 'release', 'beta'),
+        (['x86_64', 'ppce64le'], ['x86_64'], 'release', 'beta'),
         (['x86_64', 'ppce64le', 'arm64'], ['x86_64', 'ppce64le', 'arm64'], 'beta', 'beta'),
+        # pulp composes have the beta signing intent but the unsigned intent overrides that
         (['x86_64', 'ppce64le', 'arm64'], ['x86_64', 'ppce64le', 'arm64'], 'unsigned', 'unsigned'),
+        # For the next test, since arches is none, no compose is performed even though pulp_arches
+        # has a value. Expected intent doesn't change when nothing is composed.
         (['x86_64', 'ppce64le', 'arm64'], None, 'beta', 'beta'),
     ))
     @pytest.mark.parametrize(('flags', 'expected_flags'), [
         ({}, None,),
         ({UNPUBLISHED_REPOS: False}, None),
         ({UNPUBLISHED_REPOS: True, 'some_flag': True}, [UNPUBLISHED_REPOS])
     ])
+    @pytest.mark.parametrize(('multilib'), [
+        False,
+        True
+    ])
     def test_request_pulp_and_multiarch(self, workflow, reactor_config_map, pulp_arches, arches,
-                                        signing_intent, expected_intent, flags, expected_flags):
+                                        signing_intent, expected_intent, flags, expected_flags,
+                                        multilib):
         content_set = ''
         pulp_composes = {}
         base_repos = ['spam', 'bacon', 'eggs']
         pulp_id = ODCS_COMPOSE_ID
+        arches = arches or []
+        temp_arches = arches
+        pulp_arches = pulp_arches or []
+        if multilib:
+            if 'x86_64' in pulp_arches and 'i686' not in pulp_arches:
```

Comment:
Rather than repeating this logic, is there a way to include the full list of expected architectures in the parametrization?

Reply:
It's not pretty, but sure.